src/DockManager.cpp

```diff
@@ -823,6 +824,23 @@ void CDockManager::loadPerspectives(QSettings& Settings)
 	Settings.endArray();
 }
 
+CDockWidget* CDockManager::centralWidget()
+{
+    return d->CentralWidget;
+}
+
+//============================================================================
+CDockAreaWidget* CDockManager::setCentralWidget(CDockWidget* widget)
+{
+    if(d->CentralWidget)
+    {
+        addDockWidget(RightDockWidgetArea, d->CentralWidget);
+    }
+
+    d->CentralWidget = widget;
+    return addDockWidget(CenterDockWidgetArea, widget);
```

Comment:
A central widget should be not closable, not flotable and not movable and the dock area title bar should be hidden. In your example you set all these properties outside before you call this function.

What do you think about settings these properties here inside. That means, if someone sets a central widget, this function clears the movable, floatable and closable features and hides the title bar of the created dock area?